airflow/providers/amazon/aws/operators/emr.py

```diff
@@ -901,36 +930,63 @@ def execute(self, context: Context) -> str | None:
             raise AirflowException(f"Application Creation failed: {response}")
 
         self.log.info("EMR serverless application created: %s", application_id)
-
-        # This should be replaced with a boto waiter when available.
-        waiter(
-            get_state_callable=self.hook.conn.get_application,
-            get_state_args={"applicationId": application_id},
-            parse_response=["application", "state"],
-            desired_state={"CREATED"},
-            failure_states=EmrServerlessHook.APPLICATION_FAILURE_STATES,
-            object_type="application",
-            action="created",
-            countdown=self.waiter_countdown,
-            check_interval_seconds=self.waiter_check_interval_seconds,
-        )
+        attempt = 0
+        waiter = self.hook.get_waiter("serverless_app_created")
+        while attempt < self.waiter_max_attempts:
```

Comment:
Is not the purpose of a waiter to handle these different attempts? Why are you calling `waiter.wait(` with `"MaxAttempts": 1,` `self.waiter_max_attempts` times?

Reply:
it's the workaround we agreed on to be able to log the status changes while waiting. Otherwise, the waiters are completely silent until they succeed/fail.

Reply:
What do you mean by making them verbose? We can't make any changes to the waiters because they are in the botocore repo.

Reply:
Oh! Sorry, I thought you were using custom waiters

Reply:
I see ... I just feel we are going backwards. We initially decided to start using boto3 custom waiters to get rid of these while loop and to have a code cleaner. Now, we want to keep using boto3 customer waiter but wrap them around the exact same code we tried to remove initially. I now I dont provide any solution here but should we try to find another solution? If I am the only one thinking that way, I am fine

Reply:
I get what you're saying, and this is certainly not an ideal use of the boto waiters. Ideally, the waiters would allow logging natively, and we would end up with a very clean way to wait while also getting information about the current status of the resources. Unfortunately, until botocore supports that, we have to manually work to get logging. 

That being said, there is still benefit in using custom boto waiters. We get to make use of botocore's code that allows us to create new waiters without writing additional code. This is especially important with deferrable operators because aiobotocore can use the same config file to generate an async waiter without us having to write async code. 

Although we don't get to enjoy the full benefit of using boto waiters, we do gain something, which I think makes it worth doing.

Reply:
I see. I agree with you, we definitely need these logs and I am just sad this is the only way to get these. Especially because it worsen the code. My understanding is the waiter is pushing logs but at the DEBUG level right? Did you try to increase the specific logger that is sending these logs? Then, instead of pushing these logs to DEBUG level, it would do it, say, at INFO level? See [here](https://github.com/boto/boto3/issues/521) more informations.

Reply:
No the waiter is not pushing any log messages besides a debug message on completion, which is not really helpful.

Reply:
I see. I have another proposal (yes as you can see I really hate this code). Why dont you move this code in a separate function and you use `tenacity` to handle the retry. The code would look like this:

```python
@tenacity.retry(
    retry=tenacity.retry_if_exception_type(WaiterError),
    stop=tenacity.stop_after_attempt(self.waiter_max_attempts),
    wait=tenacity.wait_fixed(self.waiter_delay),
)
def wait(waiter: Waiter, application_id: str):
    try:
        waiter.wait(
            applicationId=application_id,
            WaiterConfig=prune_dict(
                {
                    "Delay": self.waiter_delay,
                    "MaxAttempts": 1,
                }
            ),
        )
    except WaiterError as error:
        if "terminal failure" in str(error):
            raise AirflowException(f"Serverless Application Creation failed: {error}")
        self.log.info(
            "Serverless Application status is: %s - %s",
            error.last_response["application"]["state"],
            error.last_response["application"]["stateDetails"],
        )
        raise
```

I am still not a big fan of this code but this is definitely better than the while loop